### DISCUSSION TITLE
Block Sidebar with black transparent background on cell phone

### DISCUSSION THREAD

**Discussion by carlospccarvalho** [unovue/shadcn-vue#920] [Q&A]
I copied Dashboard 2 from the sidebar block but it's showing up with a transparent black background on mobile. Is the block code wrong? @zernonia

**Comment by rasensio** [comment]
Same issue here 
![Screenshot 2024-12-30 at 5 54 35 PM](https://github.com/user-attachments/assets/cf9f2d4d-d0ca-473b-84fc-82a3f2f86013)

**Reply by rasensio** [reply]
Looks like is a dialog when in mobile 
`....<div data-dismissable-layer="" id="" role="dialog" aria-describedby="radix-vue-dialog-description-v-0-1" aria-labelledby="radix-vue-dialog-title-v-0-0" data-state="open" data-sidebar="sidebar" data-mobile="true" tabindex="-1" class="fixed z-50 gap-4 shadow-lg transition ease-in-out data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:duration-300 data-[state=open]:duration-500 inset-y-0 left-0 h-full border-r data-[state=closed]:slide-out-to-left data-[state=open]:slide-in-from-left sm:max-w-sm w-[--sidebar-width] bg-sidebar p-0 text-sidebar-foreground [&amp;>button]:hidden" style="--sidebar-width: 18rem; pointer-events: auto;"><div class="flex h-full w-full flex-col"><div data-sidebar="header" class="flex flex-col gap-2 p-2"><!--  --><ul data-sidebar="menu" class="flex w-full min-w-0 flex-col gap-1"><li data-sidebar="menu-item" class="group/menu-item relative"><div class="h-10"><!--v-if--><img height="50" alt="Logo" data-nuxt-img="" srcset="/_ipx/h_50/img/logo-dark.svg 1x, /_ipx/h_100/img/logo-dark.svg 2x" class="" src="/_ipx/h_50/img/logo-dark.svg"><!-- <NuxtImg src="/img/iso.svg" class="block md:hidden" alt="Logo" :height="height"/> --></div><!-- ...`

**Reply by rasensio** [reply]
For now I fixed by adding ` class="md:bg-transparent bg-gray-900"` to SidebarHeader and SidebarContent components.

**Reply by PoulLorca** [reply]
Same here. The only way to solve the problem and achieve the expected behavior was by applying the classes md:bg-transparent and bg-background in the header, content, and footer. However, using these classes in the Sidebar component didn't work, even when adding !important

**Reply by xmarkurion** [reply]
Just to drop my 5 cent's silver platter. Thanks @PoulLorca it worked. I've applied .sidebar to header content and footer and it worked well 👍🏻 .
```html
<style scoped>
.sidebar {
    @apply md:bg-transparent bg-background;
}
</style>

<SidebarHeader class="sidebar">
<SidebarContent class="sidebar">
<SidebarFooter class="sidebar">
```

**Comment by Whbbit1999** [comment]
Some css variables are not loaded. Check the latest documentation. And add them.

**Reply by x1y9** [reply]
Same issue here, where is the lastest documentation? should I add the variables manually?

**Reply by Whbbit1999** [reply]
<img width="765" alt="image" src="https://github.com/user-attachments/assets/f1aff878-5fa5-4ad0-a410-f6044425271c" />

You can try it😎
